rfc/002-extension-content-configuration-processing.md

```diff
@@ -0,0 +1,197 @@
+
+![System Context Overview](assets/extension-content-configuration-processing/system-context.drawio.png)
+
+## Design Drivers
+
+- The `ContentConfiguration` process need to be asynchronous, at the time of the web request the processed configugration must be available.
+- The `ContentConfiguration` processing needs to allow for the possibility to reject a configuration change. In case a Extension Provider responds with a invalid `ContentConfiguration` definition, the `Content Configuration` processing must be able to reject the change.
+- The `ContentConfiguration` processing to allow for inline definition of `ContentConfiguration` content as well as remote `ContentConfiguration`'s to be downloaded at time of processing
+
+## Out of Scope
+
+- This document does not detail out specifics of extensions in general, like extension classes and extension instances.
+- This document does not detail out specifics of the CDM content, and how a CDM is structured.
+
+## Options Considered
+
+- Async processing of `ContentConfiguration` using a subroutine in the Extension Manager
+
+### Async processing of ContentConfiguration using a subroutine in the Extension Manager
+
+The Extension Manager will provide a custom resource definition - Content Configuration - that allows extension providers to describe how their extension should be integrated into the portal in a declarative way. The Extension Manager will then process the CC and store the observed state in its database. The Portal, when rendering the UI will then query for these content configuration resources and consider the processed information from the status.
+
+The design of the `ContentConfiguration` should be as follows:
+
+```yaml
+apiVersion: core.openmfp.io/v1alpha1
+kind: ContentConfiguration
+metadata:
+  name: example-extension
+  namespace: default
+  generation: 1
+spec:
+  remoteConfiguration:
+    # The remote configuration can be a remote yaml or json file. 
+    url: https://example.com/example-extension-configuration.yaml
+    authentication:
+      type: basic # none (default), client credentials, bearer, basic
+      secretRef: 
+        # The Secret must contain respective keys depending on the chosen authorization type
+        # expected keys: 
+        # - username (basic) 
+        # - password (basic)
+        # - clientid (client credentials)
+        # - clientsecret (client credentials)
+        # - token (token)
+        name: example-extension-credentials 
+  # Below is just an example of a content configuration, the actual structure will be covered in a separate RFC.
+  # The extension provider must provide either a remoteConfiguration OR a inlineConfiguration
```

Comment:
Which logic should we follow if user creates `ContentConfiguration` with both remote and inline configurations?

Reply:
I think it is either/or. Both should never be configured.

Reply:
I would go with priority, if remote is configured, we ignore inline one. Or the other way around.

Reply:
for now i'd say take the inline if provided otherwise take remote.

Down the road we can have a validatingwebhook that rejects ContentConfigurations where both is specified.